plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/TestIcebergMetadataFileOperations.java

```diff
@@ -75,90 +86,238 @@ protected DistributedQueryRunner createQueryRunner()
+                        .addCopies(new FileOperation(MANIFEST, INPUT_FILE_GET_LENGTH), 4)
+                        .addCopies(new FileOperation(MANIFEST, INPUT_FILE_NEW_STREAM), 4)
+                        .addCopies(new FileOperation(METADATA_JSON, INPUT_FILE_NEW_STREAM), 1)
+                        .addCopies(new FileOperation(SNAPSHOT, INPUT_FILE_GET_LENGTH), 1)
+                        .addCopies(new FileOperation(SNAPSHOT, INPUT_FILE_NEW_STREAM), 1)
+                        .build());
+    }
+
+    @Test
+    public void testShowStatsForTableWithFilter()
+            throws Exception
+    {
+        assertUpdate("CREATE TABLE test_show_stats_with_filter AS SELECT 2 as age", 1);
+
+        assertFileSystemAccesses("SHOW STATS FOR (SELECT * FROM test_show_stats_with_filter where age >= 2)",
+                ImmutableMultiset.builder()
+                        .addCopies(new FileOperation(MANIFEST, INPUT_FILE_GET_LENGTH), 4)
+                        .addCopies(new FileOperation(MANIFEST, INPUT_FILE_NEW_STREAM), 4)
+                        .addCopies(new FileOperation(METADATA_JSON, INPUT_FILE_NEW_STREAM), 1)
+                        .addCopies(new FileOperation(SNAPSHOT, INPUT_FILE_GET_LENGTH), 1)
+                        .addCopies(new FileOperation(SNAPSHOT, INPUT_FILE_NEW_STREAM), 1)
+                        .build());
+    }
+
+    private void assertFileSystemAccesses(@Language("SQL") String query, Multiset<Object> expectedAccesses)
+            throws Exception
+    {
+        resetCounts();
+        getDistributedQueryRunner().executeWithQueryId(TEST_SESSION, query);
+        assertThat(ImmutableMultiset.<Object>copyOf(getOperations())).containsExactlyInAnyOrderElementsOf(expectedAccesses);
+    }
+
+    private static void resetCounts()
+            throws Exception
+    {
+        BEAN_SERVER.invoke(new ObjectName(BEAN_NAME), "reset", null, null);
+    }
+
+    @SuppressWarnings("unchecked")
+    private static Multiset<FileOperation> getOperations()
+            throws Exception
+    {
+        return ((Map<OperationContext, Integer>) BEAN_SERVER.getAttribute(new ObjectName(BEAN_NAME), "OperationCounts"))
+                .entrySet().stream()
+                .flatMap(entry -> nCopies(entry.getValue(), new FileOperation(entry.getKey())).stream())
+                .collect(Collectors.toCollection(HashMultiset::create));
+    }
+
+    static class FileOperation
+    {
+        private final FileType fileType;
+        private final OperationType operationType;
+
+        public FileOperation(FileType fileType, OperationType operationType)
         {
-            this.allCounts = requireNonNull(allCounts, "allCounts is null");
+            this.fileType = requireNonNull(fileType, "fileType is null");
+            this.operationType = requireNonNull(operationType, "operationType is null");
         }
 
-        public OperationCounts forQueryId(String queryId)
+        public FileOperation(OperationContext operationContext)
```

Comment:
move this above the "effective" constructor (`public FileOperation(FileType fileType, OperationType operationType)`)